src/signatures/schnorr.rs

```diff
@@ -70,6 +107,63 @@ where
         Ok(Self::new(public_nonce, s))
     }
 
+    /// Signs a message with the given secret key.
+    ///
+    /// This method correctly binds a nonce and the public key to the signature challenge, using domain-separated
+    /// hashing. The hasher is also opinionated in the sense that Blake2b 256-bit digest is always used.
+    ///
+    /// it is possible to customise the challenge by using [`construct_domain_separated_challenge`] and [`sign_raw`]
+    /// yourself, or even use [`sign_raw`] using a completely custom challenge.
+    pub fn sign_message<'a, B>(secret: &'a K, message: B) -> Result<Self, SchnorrSignatureError>
```

Comment:
Since aggregate signatures like the kernel excess need to commit to the public nonces beforehand, the API would be more useful if the public nonce is passed in. 

`wasm32-unknown-unknown` also can't use any random functions, so this method cannot be compiled to it. Passing in the public nonce would help in this case too

Reply:
So this means any application built on that target is responsible for handling its own randomness externally in general?

Reply:
Yes. So in the case of `wasm32-unknown-unknown`, the WASM VM has no RNG. You can import a native function to provide it, depending on how you are hosting the VM

Reply:
Would it then make sense to have this function also take an `Option<K>`, so the caller can provide its own nonce if required/desired? Or is that kind of fine-grained control over targets not possible within a function?

Reply:
tari_crypto uses `getrandom/js` to provide a RNG inside wasm, so these methods do work.

In any event, `pub fn sign_with_nonce_and_message` is what you're looking for.